lib/readline.nit

```diff
@@ -0,0 +1,63 @@
+# This file is part of NIT (http://www.nitlanguage.org).
+#
+# Copyright 2016 Frédéric Vachon <fredvac@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# GNU readline library wrapper
+module readline is ldflags "-lreadline"
+
+in "C" `{
+	#include <readline/readline.h>
+	#include <readline/history.h>
+`}
+
+private fun native_readline(message: NativeString): NativeString `{
+	char *line_read = readline(message);
+
+	// End of file
+	if (line_read == NULL) return NULL;
+
+	return line_read;
+`}
```

Comment:
You should clean up this method so it holds a single line of C code: `return readline(prompt);`